### PR TITLE
fix: DLT-1710 unexpected emoji on links

### PR DESCRIPTION
**Pull request by juliodialpad**
# Fix unexpected emoji on links

## Obligatory GIF (super important!)

Obligatory GIF

## :hammer_and_wrench: Type Of Change

These types will increment the version number on release:

- [x] Fix

## :book: Jira Ticket

https://dialpad.atlassian.net/browse/DLT-1710

## :book: Description

- Fixed a warning pnpm related to typescript version.
- Updated the emoji regex to detect if the emoji is at the start of the line or has a space before.
- Updated components that use emojiRegex

## :bulb: Context

If a string such as a URL contained a valid emoji shortcode it'd be replaced but that's not intended.

## :pencil: Checklist

For all PRs:

- [x] I have ensured no private Dialpad links or info are in the code or pull request description (Dialtone is a public repo!).
- [x] I have reviewed my changes.

For all Vue changes:

- [x] I have made my changes in Vue 2 and Vue 3. Note: you may sync your changes from Vue 2 to Vue 3 (or vice versa) using the `./scripts/dialtone-vue-sync.sh` script. Read docs here: [Dialtone Vue Sync Script](../packages/dialtone-vue3/.github/CONTRIBUTING.md#dialtone-vue-sync-script)

## :camera: Screenshots / GIFs

Before
Screenshot 2024-05-09 at 12 34 41 p m

After
Screenshot 2024-05-09 at 12 34 06 p m

### CODE REVIEW

**Comment by braddialpad** [comment]
Looks like some test / deploy preview failures.

**Comment by juliodialpad** [comment]
> Looks like some test / deploy preview failures.

It's due to some issue with slash commands on Vue 3, fixing

File: packages/dialtone-vue2/common/emoji/index.js [hunk 1]
```diff
@@ -15,6 +15,8 @@ export let emojiFileExtensionLarge = '.png';
 
 export const emojiJson = emojiJsonLocal;
 
+export const emojiShortCodeRegex = /(^| ):\w+:/g;
```
Comment (braddialpad) [inline packages/dialtone-vue2/common/emoji/index.js]: It's a pretty common case to put multiple emojis beside each other without spaces like so :smile::smile::smile::smile:. Maybe we can do a special case to not escape when there is a "double colon"

Comment (braddialpad) [reply packages/dialtone-vue2/common/emoji/index.js]: Also if you use a unicode emoji in emoji-text-wrapper it allows you to do this, but not with shortnames

File: packages/dialtone-vue2/components/rich_text_editor/extensions/emoji/emoji.js [hunk 2]
```diff
@@ -1,29 +1,29 @@
...
 import { PluginKey } from '@tiptap/pm/state';
 
 import Suggestion from '@tiptap/suggestion';
 import suggestionOptions from './suggestion';
 
 export const EmojiPluginKey = new PluginKey('emoji');
 
-const inputShortCodeRegex = /:\w+:$/;
-export const emojiShortCodeRegex = /:\w+:/g;
+const inputShortCodeRegex = /(^| )(:\w+:)$/;
```
Comment (braddialpad) [inline packages/dialtone-vue2/components/rich_text_editor/extensions/emoji/emoji.js]: Is it necessary to have this? Maybe we can just make our const a regex string `"(^| )(:\w+:)"` and then append a $ to it here? then use `new RegExp` to turn the string into a regex? Not that big of deal though.

File: packages/dialtone-vue2/common/emoji/index.js (hunk 1 above)
Comment (juliodialpad) [reply packages/dialtone-vue2/common/emoji/index.js]: > It's a pretty common case to put multiple emojis beside each other without spaces like so 😄😄😄😄. Maybe we can do a special case to not escape when there is a "double colon"

That implied the use of `Positive Lookbehind` which IIRC we just had an issue with it on older iOS devices.

Comment (juliodialpad) [reply packages/dialtone-vue2/common/emoji/index.js]: > Also if you use a unicode emoji in emoji-text-wrapper it allows you to do this, but not with shortnames

It's using the same emojiShortCodeRegex so this should be fixed too.

File: packages/dialtone-vue2/components/rich_text_editor/extensions/emoji/emoji.js (hunk 2 above)
Comment (juliodialpad) [reply packages/dialtone-vue2/components/rich_text_editor/extensions/emoji/emoji.js]: > Is it necessary to have this? Maybe we can just make our const a regex string `"(^| )(:\w+:)"` and then append a $ to it here? then use `new RegExp` to turn the string into a regex? Not that big of deal though.

Yes, most places use `emojiShortCodeRegex` which already includes a global flag.

That's the only place where we need `$` at the end.

I know is not ideal but seems better that having a string and having to call `new RegExp()` everywhere adding the `g` flag.

**Review by iropolo — APPROVED** [review]

File: packages/dialtone-vue2/common/emoji/index.js (hunk 1 above)
Comment (braddialpad) [reply packages/dialtone-vue2/common/emoji/index.js]: > That implied the use of `Positive Lookbehind` which IIRC we just had an issue with it on older iOS devices.

hmm wasn't that negative lookbehind?

**Review by braddialpad — APPROVED** [review]

File: packages/dialtone-vue2/common/emoji/index.js (hunk 1 above)
Comment (juliodialpad) [reply packages/dialtone-vue2/common/emoji/index.js]: > > That implied the use of `Positive Lookbehind` which IIRC we just had an issue with it on older iOS devices.
> 
> hmm wasn't that negative lookbehind?

Yeah, I think the issues were with negative lookbehind but they both have the same compatibility. Let's just keep an eye on this if some issues come related to it.

**Pull request merged**